tools/cudaq-qpud/README.md

```diff
@@ -0,0 +1,92 @@
+debugging for `remote-mqpu` or `nvqc` targets. This file is primarily intended
+for CUDA-Q developers, not end users. See the user-facing docs here:
+
+- [`remote-mqpu`](https://nvidia.github.io/cuda-quantum/latest/using/backends/platform.html#remote-mqpu-platform)
+- [`nvqc`](https://nvidia.github.io/cuda-quantum/latest/using/backends/nvqc.html)
+
+## Fully local within `cuda-quantum-dev` container
+
+The first step is usually to run the server in a separate window from the
+client by disabling any sort of auto-launch capabilities.
+
+1. In one window, launch `cudaq-qpud --port 3030 --type nvcf`. You may also
+   prefix this command with `CUDAQ_LOG_LEVEL=info` to turn on additional
+   logging in the server.
+2.
+   - If you are using Python, change your `cudaq.set_target` line to be
+   something like this: `cudaq.set_target('remote-mqpu', url='localhost:3030')`.
+   - If you are using C++, change your `nvq++` command to something like this:
+   `nvq++ --target remote-mqpu --remote-mqpu-url localhost:3030`.
+
+Note: when you run the server with `--type nvcf`, that means that the
+`cudaq-qpud` process will shut down after every invocation, so you will have to
+manually restart it if you want to invoke it again.
+
+## Fully local testing using a true NVQC image
+
+If you want to run the server in a fully "contained" environment like it is run
+for NVQC, then you can perform the following steps.
+
+1. Build your NVQC server Docker container using this command: `docker build -t nvcr.io/pnyjrcojiblh/cuda-quantum/cuda-quantum:custom -f docker/release/cudaq.nvqc.Dockerfile .`
+2. Launch the server on your local machine: `docker run -it --rm --gpus all --network=host -e NVQC_REST_PAYLOAD_VERSION=1 -e NUM_GPUS=1 -e WATCHDOG_TIMEOUT_SEC=3600 -e RUN_AS_NOBODY=1 nvcr.io/pnyjrcojiblh/cuda-quantum/cuda-quantum:custom`
+   - Note: You need to set the environment variables as intended for your
+    environment. If you are running on a multi-GPU machine, you may
+    want to set `NUM_GPUS=4` (updating `4` to the correct number for your
+    machine) and use something like `--gpus '"device=0,1,2,4"'` (in case your
+    machine has more GPUs than you want activated with `NUM_GPUS`.)
+3. Get the IP address of your host by running `ifconfig`. For the following
+   steps, let's assume your IP address is `172.31.123.45`.
+4. Similar to step 2 in the section above, but changing `localhost` to the
+   correct IP address obtained in step 3:
+   - If you are using Python, change your `cudaq.set_target` line to be
+   something like this: `cudaq.set_target('remote-mqpu', url='172.31.123.45:3030')`.
+   - If you are using C++, change your `nvq++` command to something like this:
+   `nvq++ --target remote-mqpu --remote-mqpu-url 172.31.123.45:3030`.
+
+## Running your own image on the NVQC server
+
+1. After building your `nvcr.io/pnyjrcojiblh/cuda-quantum/cuda-quantum:custom`
+   image, you can `docker push` it (assuming you have authorized credentials).
```

Comment:
Is this section supposed to be internal-only?

Reply:
> Is this section supposed to be internal-only?

The `pnyjrcojiblh` identifier is also in our `.github/workflows/integration_tests.yml` workflow, and it certainly isn't a secret. However, I can add a sentence or two explaining this clarifying tag and providing an alternative if people want to name it something else.

Reply:
I see. I didn't mean that it was confidential, but it may not be useful to anyone external?

Reply:
> I see. I didn't mean that it was confidential, but it may not be useful to anyone external?

Yeah, I think this whole README.md file would be of limited utility to anyone external (because hopefully they don't need to debug issues like this), but IMO, it seems more convenient to place the docs near the code anyway.

Reply:
Clarified in 106b8e9e1